src/main/java/mods/railcraft/client/particles/ParticleHelper.java

```diff
@@ -89,23 +96,25 @@ public static boolean addHitEffects(World world, Block block, MovingObjectPositi
      */
     @SideOnly(Side.CLIENT)
     public static boolean addDestroyEffects(World world, Block block, BlockPos pos, IBlockState state, EffectRenderer effectRenderer, ParticleHelperCallback callback) {
-        if (block != WorldPlugin.getBlock(world, pos)) return true;
-        byte its = 4;
-        for (int i = 0; i < its; ++i) {
-            for (int j = 0; j < its; ++j) {
-                for (int k = 0; k < its; ++k) {
-                    double px = x + (i + 0.5D) / (double) its;
-                    double py = y + (j + 0.5D) / (double) its;
-                    double pz = z + (k + 0.5D) / (double) its;
-                    int random = rand.nextInt(6);
-                    EntityDiggingFX fx = new EntityDiggingFX(world, px, py, pz, px - x - 0.5D, py - y - 0.5D, pz - z - 0.5D, block, random, state);
-                    fx.setParticleIcon(block.getIcon(world, x, y, z, 0));
-                    if (callback != null)
-                        callback.addDestroyEffects(fx, world, pos, state);
-                    effectRenderer.addEffect(fx.func_174845_l());
-                }
-            }
-        }
+//        if (block != WorldPlugin.getBlock(world, pos)) return true;
+        // TODO 1.8.9 port make sure this si the correct function and it works!
+        effectRenderer.addBlockDestroyEffects(pos, state);
+//        byte its = 4;
```

Comment:
Then it would be better to remove the override at the block level. But I still doubt that it will use the correct texture and yes there are ways to access textures directly from what I saw in my brief inspection of this code.